documentation/projects/proposals/trust_and_safety/detecting_sensitive_textual_content/20230506-implementation_plan_frontend_blurring_sensitive.md

```diff
@@ -0,0 +1,264 @@
+
+### Step 2: Feature flag system enhancements
+
+The feature flag + preferences mechanism will need some new features to support
+some of the new use cases proposed for this feature.
+
+- Ability to store and retrieve preferences from `sessionStorage` instead of
+  `localStorage`.
+- Option to not support `ff_` query parameters for a particular flag.
+
+These features are needed to support the new feature flag
+[`fetch_sensitive`](#fetch_sensitive).
+
+### Step 3: New feature flags
+
+```{caution}
+Observe the difference between on/off and enabled/disabled. The former refers to
+the position/value of the toggle, while the latter refers to whether the toggle
+can be interacted with and flipped.
+```
+
+Two feature flags need to be developed for this feature. One related feature
+flag, `fake_sensitive`, already exists.
+
+#### `fetch_sensitive`
+
+This is not technically a feature flag, but rather a preference, which is
+internally the same API. As a preference, it will always be set to 'switchable'
+in all environments, defaulting to 'off', and will be controlled by the user via
+the switches in the search results sidebar.
+
+This flag will not support the use of `ff_` query parameters, to prevent
+URL-based malicious action.
+
+#### `sensitive_content`
+
+This is the main, real feature flag that enables the entire feature of being
+able to get (or avoid) mature content and then see or blur it. This flag will
+determine whether the two switches even appear in the sidebar.
+
+It will be set to 'enabled' in development, 'switchable' in staging and
+'disabled' in production till launch, post which it will be 'enabled'
+everywhere.
+
+#### What about the blur toggle?
+
+The second toggle in the mockups that determines whether to blur the sensitive
+results (let's refer to as `blur_sensitive`) is an ephemeral toggle that is
+off + disabled when `fetch_sensitive` is off and enabled only when
+`fetch_sensitive` is on.
+
+It will be stored in Pinia so that it can be accessed by the components of
+search result cells and blur button in single result view. The UI store would be
+a good place to store the value of this toggle. However, unlike other values in
+the UI store, it would not be persisted to a cookie and instead would be reset
+on a hard refresh.
+
+Note that the hide button that appears on an unblurred single result page is
+gone if the user has completely disabled blurring from the search sidebar.
+
+### Step 3: Updating search store
+
+In this step we will update the search mechanism to add support for the
+`include_sensitive_results` parameter, which supersedes the existing `mature`
+parameter. The list of search results will include mature content if this is set
+to `true`.
+
+Since the API will also send the `sensitivity` field, the types for media items
+will need to be updated to account for that. This will boil down to a boolean
+value of `is_sensitive` for each media item.
```

Comment:
I'll delete this if it covered later, but the single results page design shows different copy depending on the values in the sensitivity array.

Reply:
Fixed to reflect dual purpose of `sensitivity` in 7d2e993.